### PR TITLE
Added Exponent Finding program

### PR DESCRIPTION
**Pull request by Utsav1999**
### **Describe your change:**
This Python recursion program is for finding exponent.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Utsav1999** [comment]
**The command "flake8 --ignore=E203,W503 --max-complexity=25 --max-line-length=88 --statistics --count ." failed and exited with 1 during .**

How to resolve this?

**Comment by lance-pyles** [comment]
The issues are listed.  Go to details then The build link and scroll down.  Looks like formatting issues mostly.

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/blob/master/Travis_CI_tests_are_failing.md

**Comment by vasugamdha** [comment]
I can help you to pass these checks but i don't know how to help and edit the code in this PR? @Utsav1999

**Comment by cclauss** [comment]
Autofix with `python3 -m pip install black ; black maths/exponent_recursion.py`

File: maths/exponent_recursion.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        3 to the power of 4: 81
+
+
+        i/p -->
+        Enter the base: 2
+        Enter the exponent: 0
+        o/p -->
+        2 to the power of 0: 1
+'''
+def powerCalculation(n: int, p: int)->int:
```
Comment (cclauss) [inline maths/exponent_recursion.py:17]:
```suggestion
def power(base: int, exponent: int)->int:
    """
    >>> all(power(base, exponent) == pow(base, exponent)
    ...     for base in range(10) for exponent in range(10))
    True
    """
    return base * power(n, p - 1) if p > 1 else 1
```

**Pull request closed without merging**